app/gui/src/dashboard/App.tsx

```diff
@@ -594,5 +592,49 @@ function LocalBackendPathSynchronizer() {
       localBackend.resetRootPath()
     }
   }
+
   return null
 }
+
+/** Subscribe to queries to keep them up to date. */
+function QuerySubscriber() {
```

Comment:
What's the purpose on having this in the App.tsx?

Reply:
doesn't have to be - i just wanted to lift it up out of... drive or whatever which i felt was a weird place to have it

Reply:
I still don't get why we extracted this into a separate component. What was the purpose of that? Do we need to refetch directories on `any` page? even if user not logged in? on subscribe page?